pkg/controller/shared_informer.go

```diff
@@ -0,0 +1,101 @@
+package controller
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+
+	oclient "github.com/openshift/origin/pkg/client"
+)
+
+type InformerFactory interface {
+	Start(stopCh <-chan struct{})
+
+	Pods() PodInformer
+}
+
+type PodInformer interface {
+	Informer() framework.SharedIndexInformer
+	Indexer() cache.Indexer
```

Comment:
I would prefer to get these from the informer since it already has them and we don't need to maintain different interfaces. It seems that the underlying controller implementations are responsible for boiling their interfaces down to what they need.